services/storage/src/simcore_service_storage/db_file_meta_data.py

```diff
@@ -27,10 +27,12 @@ async def exists(conn: SAConnection, file_id: SimcoreS3FileID) -> bool:
     )
 
 
-async def upsert(conn: SAConnection, fmd: FileMetaData) -> FileMetaDataAtDB:
+async def upsert(
+    conn: SAConnection, fmd: Union[FileMetaData, FileMetaDataAtDB]
+) -> FileMetaDataAtDB:
     # NOTE: upsert file_meta_data, if the file already exists, we update the whole row
     # so we get the correct time stamps
-    fmd_db = FileMetaDataAtDB.from_orm(fmd)
+    fmd_db = FileMetaDataAtDB.from_orm(fmd) if isinstance(fmd, FileMetaData) else fmd
```

Comment:
THOUGHT: what do you intent with this ''if" check ... saving a copy? 
NOTE that ``fmd`` is a reference (that could be used outside) and ``fmd_db`` could eventually be modified and therefore the latter as well ... which could be dangerous. Again, I would prioritize safety to performance

Reply:
on second thought, I think I can keep it. 
I use from_orm to get the FileMetaDataAtDB type in order to upsert in the database. If I get directly a FileMetaDataAtDB type as argument then I directly use it. I think that is ok. And I return the one inserted which is anyway another instance. I think that it correct right?